src/ucp/tag/rndv.c

```diff
@@ -1324,29 +1641,30 @@ UCS_PROFILE_FUNC(ucs_status_t, ucp_rndv_rtr_handler,
                       ucp_ep_peer_name(ep), ucs_status_string(status));
         }
 
-        is_pipeline_rndv = ((!UCP_MEM_IS_ACCESSIBLE_FROM_CPU(sreq->send.mem_type) ||
-                             (sreq->send.length != rndv_rtr_hdr->size)) &&
-                            (context->config.ext.rndv_mode != UCP_RNDV_MODE_PUT_ZCOPY));
+        is_put_pipeline_rndv = ((!UCP_MEM_IS_ACCESSIBLE_FROM_CPU(sreq->send.mem_type) ||
+                                (sreq->send.length != rndv_rtr_hdr->size)) &&
+                                (context->config.ext.rndv_mode != UCP_RNDV_MODE_PUT_ZCOPY));
 
         sreq->send.lane = ucp_rkey_find_rma_lane(ep->worker->context, ep_config,
-                                                 (is_pipeline_rndv ?
+                                                 (is_put_pipeline_rndv ?
                                                   sreq->send.rndv_put.rkey->mem_type :
                                                   sreq->send.mem_type),
                                                  ep_config->tag.rndv.put_zcopy_lanes,
                                                  sreq->send.rndv_put.rkey, 0,
                                                  &sreq->send.rndv_put.uct_rkey);
+
         if (sreq->send.lane != UCP_NULL_LANE) {
             /*
              * Try pipeline protocol for non-host memory, if PUT_ZCOPY protocol is
              * not explicitly required. If pipeline is UNSUPPORTED, fallback to
```

Comment:
this comment does not seem relevant now

Reply:
this is still relavant for D->H and D->D case